src/GitHubExtension/Widgets/GitHubUserWidget.cs

```diff
@@ -57,17 +58,44 @@ public override void OnWidgetContextChanged(WidgetContextChangedArgs contextChan
 
     protected override void ResetWidgetInfoFromState()
     {
+        JsonNode? dataObject = null;
+
         try
         {
-            var dataObject = JsonNode.Parse(ConfigurationData);
+            dataObject = JsonNode.Parse(ConfigurationData);
+        }
+        catch (JsonException)
+        {
+            // Old data versioning was not a Json string. If we attempt to parse
+            // and we get a failure, assume it is the old version.
+            if (!string.IsNullOrEmpty(ConfigurationData))
+            {
+                Log.Logger()?.ReportInfo(Name, ShortId, $"Found string data format, migrating to JSON format. Data: {ConfigurationData}");
+                var migratedState = new JsonObject
+                {
+                    { "showCategory", ConfigurationData },
+                };
 
-            if (dataObject == null)
+                // Prior to this configuration change, multi-account was not supported. Assume that
+                // if there is exactly one Developer Id that is the one the user had configured.
+                if (DeveloperIdProvider.GetInstance().GetLoggedInDeveloperIds().DeveloperIds.Count() == 1)

```

Comment:
Nit: A comment here about why an "else" case isn't expected would be nice.

Reply:
Sure, I'll add. I verified that the else case isn't needed because an empty developer selection punts the widget back to the configuratin page where the user can select the developer ID they want.